synapseadmin/roomapi.go

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2023 Tulir Asokan
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package synapseadmin
+
+import (
+	"context"
+	"net/http"
+	"maunium.net/go/mautrix"
+	"maunium.net/go/mautrix/id"
+	"strconv"
+)
+
+type ReqListRoom struct {
+	Alias	string
+	From	int
+	Limit   int
+}
+
+func (cli *Client) NewReqListRoom() ReqListRoom {
+   req := ReqListRoom{}
+   req.Alias = ""
+   req.From = 0
+   req.Limit = 100
+   return req
+}
+
+func (req *ReqListRoom) BuildQuery() map[string]string {
+	query := map[string]string{
+		"limit": strconv.Itoa(req.Limit),
+	}
+	if req.Alias != "" {
+		query["search_term"] = req.Alias
+	}
+	query["from"] = strconv.Itoa(req.From)
+	return query
+}
+
+//  Get room info based on alias
+//
+// https://matrix-org.github.io/synapse/latest/admin_api/rooms.html
+func (cli *Client) ListRoom(ctx context.Context, req ReqListRoom) (*mautrix.RoomsResponse ,error) {
+	var resp mautrix.RoomsResponse
+	var reqURL string
+	reqURL = cli.BuildURLWithQuery(mautrix.SynapseAdminURLPath{"v1", "rooms"}, req.BuildQuery())//map[string]string{"search_term": alias})
+	_, err := cli.MakeFullRequest(ctx, mautrix.FullRequest{
+		Method:       http.MethodGet,
+		URL:          reqURL,
+		ResponseJSON: &resp,
+	})
+	return &resp, err
+}
+
+// ReqDeleteRoom is the request content for Client.DeleteRoom.
+type ReqDeleteRoom struct {
+	// Room Id to delete
+	RoomId string `json:"room_id"`
```

Comment:
Initialisms like `URL` and also `ID` (when short for "identifier") should be consistent in case (i.e. `id` or `ID`, never `Id`) https://go.dev/wiki/CodeReviewComments#initialisms

Also, there are some primitive types in the `id` package which should always be used when applicable https://pkg.go.dev/maunium.net/go/mautrix/id#RoomID

(both of these apply to multiple places in the code)